transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java

```diff
@@ -422,6 +433,10 @@ public void connect(SocketAddress remote, SocketAddress local, ChannelPromise ch
                     KQueueDatagramChannel.this.local = socket.localAddress();
                     success = true;
 
+                    // Need to be set before we notify the promise and fire through the pipeline.
+                    active = true;
```

Comment:
For epoll this method was mostly moved into the base class (`AbstractEpollChannel`). Can we keep the code consistent here too and move most of this code into the base class (`AbstractKQueueChannel`)? It will make future maintenance much easier to keep epoll/kqueue as consistent as possible.